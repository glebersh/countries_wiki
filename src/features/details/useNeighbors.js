import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { useSelector } from "react-redux";
import { loadNeighborsByBorder, selectNeighbors } from "./detailsSlice";



export const useNeighbors = (borders = []) => {

  const dispatch = useDispatch();
  const neighbors = useSelector(selectNeighbors);

  useEffect(() => {
    if (borders.length) {
      dispatch(loadNeighborsByBorder(borders));
    }
  }, [borders, dispatch]);
  return neighbors;
};