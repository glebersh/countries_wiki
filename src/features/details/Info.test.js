import { render, screen } from '@testing-library/react';
import { toBeInTheDocument, toHaveBeenCalledWith } from '@testing-library/jest-dom';
import { useNavigate } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { Info } from './Info';
import { useNeighbors } from './useNeighbors';
import * as reduxHooks from 'react-redux';


const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('./useNeighbors', () => ({
  useNeighbors: () => ['Columbia', 'Argentina', 'Peru'],
}));

const data = {
  name: 'Brasil',
  nativeName: 'Brasil',
  flag: 'https://via.placeholder.com/300.png',
  capital: 'Brasilia',
  population: 214047375,
  region: 'Amerika',
  subregion: 'Latin Amerika',
  topLevelDomain: ['.br'],
  currencies: [{
    code: 'BRL',
    name: 'Brazilian real',
    symbol: 'R$'
  }],
  languages: [{
    iso639_1: 'pt',
    iso639_2: 'por',
    name: 'Portuguese',
    nativeName: 'PortuguÃªs'
  }],
  borders: ['Columbia', 'Argentina', 'Peru'],
  push: mockedUsedNavigate,
};

describe('Info component', () => {
  it('Info renders', () => {
    render(<Info {...data} />)
    expect(screen.getByTestId('country-info')).toBeInTheDocument();
  });

  it('Info snapshot', () => {
    const countryInfo = render(<Info {...data} />);
    expect(countryInfo).toMatchSnapshot();
  });

  it('Info neighbors navigate', () => {
    render(<Info {...data} />);
    userEvent.click(screen.getByText('Columbia'));
    expect(screen.getByText('Columbia')).toBeInTheDocument();
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/countries_wiki/country/Columbia');
  });
});