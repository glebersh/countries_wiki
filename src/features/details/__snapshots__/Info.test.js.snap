// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component Info snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-bcXHqe kOecqs"
      >
        <img
          alt="Brasil"
          class="sc-gswNZR iDRsIg"
          src="https://via.placeholder.com/300.png"
        />
        <div
          data-testid="country-info"
        >
          <h1
            class="sc-dkrFOg bkwYfw"
          >
            Brasil
          </h1>
          <div
            class="sc-hLBbgP kCEmcI"
          >
            <ul
              class="sc-eDvSVe dtdIsR"
            >
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Native Name:
                </b>
                 
                Brasil
              </li>
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Population
                </b>
                 
                214047375
              </li>
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Region:
                </b>
                 
                Amerika
              </li>
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Sub Region:
                </b>
                 
                Latin Amerika
              </li>
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Capital:
                </b>
                 
                Brasilia
              </li>
            </ul>
            <ul
              class="sc-eDvSVe dtdIsR"
            >
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Top Level Domain
                </b>
                 
                <span>
                  .br
                </span>
              </li>
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Currency
                </b>
                 
                <span>
                  Brazilian real
                   
                </span>
              </li>
              <li
                class="sc-jSUZER kAqNzz"
              >
                <b>
                  Top Level Domain
                </b>
                 
                <span>
                  Portuguese
                </span>
              </li>
            </ul>
          </div>
          <div
            class="sc-gKPRtg gzOaMP"
          >
            <b>
              Border Countries
            </b>
            <div
              class="sc-iBYQkv kygipb"
            >
              <span
                class="sc-ftTHYK iZdCez"
              >
                Columbia
              </span>
              <span
                class="sc-ftTHYK iZdCez"
              >
                Argentina
              </span>
              <span
                class="sc-ftTHYK iZdCez"
              >
                Peru
              </span>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-bcXHqe kOecqs"
    >
      <img
        alt="Brasil"
        class="sc-gswNZR iDRsIg"
        src="https://via.placeholder.com/300.png"
      />
      <div
        data-testid="country-info"
      >
        <h1
          class="sc-dkrFOg bkwYfw"
        >
          Brasil
        </h1>
        <div
          class="sc-hLBbgP kCEmcI"
        >
          <ul
            class="sc-eDvSVe dtdIsR"
          >
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Native Name:
              </b>
               
              Brasil
            </li>
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Population
              </b>
               
              214047375
            </li>
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Region:
              </b>
               
              Amerika
            </li>
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Sub Region:
              </b>
               
              Latin Amerika
            </li>
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Capital:
              </b>
               
              Brasilia
            </li>
          </ul>
          <ul
            class="sc-eDvSVe dtdIsR"
          >
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Top Level Domain
              </b>
               
              <span>
                .br
              </span>
            </li>
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Currency
              </b>
               
              <span>
                Brazilian real
                 
              </span>
            </li>
            <li
              class="sc-jSUZER kAqNzz"
            >
              <b>
                Top Level Domain
              </b>
               
              <span>
                Portuguese
              </span>
            </li>
          </ul>
        </div>
        <div
          class="sc-gKPRtg gzOaMP"
        >
          <b>
            Border Countries
          </b>
          <div
            class="sc-iBYQkv kygipb"
          >
            <span
              class="sc-ftTHYK iZdCez"
            >
              Columbia
            </span>
            <span
              class="sc-ftTHYK iZdCez"
            >
              Argentina
            </span>
            <span
              class="sc-ftTHYK iZdCez"
            >
              Peru
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
